src/server/components/dashboard/controllers.ts

```diff
@@ -253,10 +253,18 @@ export async function listEditAddPublisher(req: Request, res: Response, next: Ne
   if (listId === "new") {
     const newList = await createList(data);
 
-    if (newList?.id !== undefined) {
+    if ((newList as Record<string, boolean>).duplicateListError) {
```

Comment:
type narrowing is probably cleaner here, instead of having to cast a fairly chunky inline type. 

```
    const isDuplicateListError = "duplicateListError" in newList;

    if (isDuplicateListError) { // technically, this only checks if the property exists, and not if the value is true. 

    // any reference to newList in this block will be of type duplicateListError 

      return
    }


    // because isDuplicateListError has returned, all references to newList will now be `List`, so no need to cast


    
```